include/lbann/layers/loss/mean_squared_error.hpp

```diff
@@ -103,6 +103,41 @@ class mean_squared_error_layer : public data_type_layer<TensorDataType> {
   data_layout get_data_layout() const override { return T_layout; }
   El::Device get_device_allocation() const override { return Dev; }
 
+#ifdef LBANN_HAS_ONNX
+  void fill_onnx_node(onnx::GraphProto& graph) const override {
+    auto* diff = graph.add_node();
+    for(auto const* parent : this->get_parent_layers()) {
+      size_t idx = parent->find_child_layer_index(*this);
+      diff->add_input(parent->get_name() + "_" + std::to_string(idx));
+    }
+    diff->add_output("diff_0");
+    diff->set_name("diff");
```

Comment:
We should make sure that all ONNX nodes and tensors have unique names if the model has multiple MSE layers:
```suggestion
    diff->add_output(this->get_name() + "_diff_0");
    diff->set_name(this->get_name() + "_diff");
```
Similar changes should be made to the rest of the function.